### PR TITLE
Logs: Fix pagination skipping entries

### LINKED ISSUE #38681: Logs navigation: Pagination skips entries
**What happened**:

Trying to get `Older logs` skips entries.
The initial page contains entries from `21:52:58` newest entry to  `21:20:30` oldest entry (position 500), but the page time is incorrectly computed to `21:19:22` oldest entry. 
(Don't know if it's relevant, but I have duplicate entries on `21:20:30`)

Thus, when retrieving an older page, the page starts from `21:19:22` skipping results. I have multiple entries on `21:20:13` that are skipped.
image
image

Also notice that in the second page, the first entry has time `21:19:06` but the page is displayed incorrectly to be from `21:19:22`.

**What you expected to happen**:

Normal pagination.

**How to reproduce it (as minimally and precisely as possible)**:

1. Search log entries returining more than the 500 limit
2. Bottom of page, click `Older logs` and notice results are skipped

**Anything else we need to know?**:

Related to PR #33259 and feature #19354

**Environment**:
- Grafana version:  Grafana v8.1.0 (62e720c)
- Data source type & version: Elasticsearch 7.10.1
- OS Grafana is installed on: docker
- User OS & Browser: Win10 & Firefox
- Grafana plugins: default
- Others:

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

Pagination was skipping entries as we were using visibleRange.from [here](https://github.com/grafana/grafana/blob/main/public/app/features/explore/LogsNavigation.tsx#L115) (calculated by [getSeriesProperties](https://github.com/grafana/grafana/blob/main/public/app/core/logs_model.ts#L271)), not the timestamp of the oldest log entry in the page. 

This PR does 2 things: 
1. Renames `visibleRange` to `histogramRange` as it is only used in old histogram. It is calculated specifically for it and it can't be used anywhere else because of the aligning/bucketing logic. Moreover when/if we decide to remove old histogram, it is more clear that this `histogramRange` logic can be removed as well. 

1. Introduces `logsRange` which is basically just timestamp of first and last log line. Currently this is only used for pagination, but it could be used anywhere else where you need timestamp of the first and last log. I considered also calculating this directly in the Logs component, but then you need to make sure what is the ordering of the logs is correct (as they can be flipped) and this places seemed like the best place to calculate it as we have logs already sorted. 

But I am interested to hear your thoughts! 👂 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/38681

### CODE REVIEW

File: packages/grafana-data/src/types/logs.ts [hunk 1]
```diff
@@ -85,7 +85,8 @@ export interface LogsModel {
   meta?: LogsMetaItem[];
   rows: LogRowModel[];
   series?: DataFrame[];
-  visibleRange?: AbsoluteTimeRange;
+  histogramRange?: AbsoluteTimeRange;
```
Comment (gabor) [inline packages/grafana-data/src/types/logs.ts:88]: do you think we should push more the "this is the old approach" here? maybe name it `legacyHistogramRange`? or jus tname it `histogramRange` and deprecate it immediately?

Comment (ivanahuckova) [reply packages/grafana-data/src/types/logs.ts:88]: well, we didn't say that we are deprecating histogram yet. So if we want to do that, we should do it in 1 PR. But we didn't discuss doing so. 

Keep in mind that histogram is used by all other data sources that don't implement LogVolumes.

Comment (gabor) [reply packages/grafana-data/src/types/logs.ts:88]: yeah, i see your point... btw. is this something we need to put into the breaking-changes section?

Comment (ivanahuckova) [reply packages/grafana-data/src/types/logs.ts:88]: Hmmm, good point. Maybe we shouldn't rename it and just add comment.

File: public/app/features/explore/Logs.tsx [hunk 2]
```diff
@@ -163,7 +163,7 @@ class UnthemedLogs extends PureComponent<Props, State> {
     }
   };
 
-  onChangewrapLogMessage = (event: React.ChangeEvent<HTMLInputElement>) => {
+  onChangeWrapLogMessage = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (ivanahuckova) [inline public/app/features/explore/Logs.tsx:166]: unrelated spellcheck fix

File: packages/grafana-data/src/types/logs.ts (hunk 1 above)
Comment (ivanahuckova) [reply packages/grafana-data/src/types/logs.ts]: As we talked about, I have changed the approach and implemented navigationRange calculation in LogsComponent

**Review by gabor — APPROVED** [review]
LGTM!

**Pull request merged**